cloudprem/physical/util/dms-stop.sh

```diff
@@ -4,25 +4,35 @@ TRIGGER="$1"
 AWS_REGION="$2"
 AWS_PROFILE="$3"
 
-function checkForStoppedDMS() {
+function getDMSStatus() {
   aws dms describe-replication-tasks --filter Name=replication-task-arn,Values="$TRIGGER" --without-settings --region "$AWS_REGION" --profile "$AWS_PROFILE" |jq --raw-output '.[][0]["Status"]'
 }
 
+function stopDMS() {
+  local STATUS
+
+  STATUS=$(getDMSStatus)
+   if [[ "$STATUS" == "running" ]]; then
+     echo -e "Stopping DMS Task..."
+     aws dms stop-replication-task --replication-task-arn "$TRIGGER" --region "$AWS_REGION" --profile "$AWS_PROFILE" > /dev/null
+     waitforStoppedDMS
+   else
+     echo -e "DMS Already Stopped."
+   fi
+}
+
 function waitforStoppedDMS() {
   echo "Waiting for the DMS task to stop..."
 
-   local STATUS
+  local STATUS
 
-   STATUS=$(checkForStoppedDMS)
-   until [[ "$STATUS" == "stopped" ]]; do
-      sleep 10
-      STATUS=$(checkForStoppedDMS)
-   done
+  STATUS=$(getDMSStatus)
+  while [[ "$STATUS" == "stopping" ]]; do
+    sleep 10
+    STATUS=$(getDMSStatus)
+  done
 
-   echo -e "DMS Task Stopped Successfully."
+  echo -e "DMS Task Stopped Successfully."
```

Comment:
Do you want to double check that `STATUS == 'stopped'`? Is there any other states it could be here?

Reply:
There are a bunch of statuses it could be in which is why I only check for the ones that indicate it is still running, i.e. `running` and `stopping`. Any other possible status means it is stopped and we don't need to do anything else. You can see the first version checked for "stopped" and it occasionally caused an infinite loop because the DMS task might be in one of those other statuses that's effectively "stopped" but named something else like "stopped with error".